src/main-view/content/row-settings/send-keys-row-settings.js

```diff
@@ -118,33 +133,78 @@ class SendKeysRowSettings extends Gtk.Grid {
     this.onBeginEndCombo.connect('changed', () => this.emit('changed'));
     this.animationCombo.connect('changed', () => this.emit('changed'));
 
+    this.shortcutButton.connect('require_clear_shortcut', this.clearShortcut);
+    this.shortcutButton.connect('modifier_addition', (widget, key) => {
+      if (this.modifiers.includes(key)) {
+        return;
+      }
+
+      this.modifiers.push(key);
+      this.modifiersEntry.text = this.modifiers.toString().replace(/,/g, '+');
+
+      this.buildShortcutLabelContent();
+    });
+    this.shortcutButton.connect('key_addition', (widget, key, keyval) => {
+      /// if escape, clear
+      if (keyval === Gdk.KEY_Escape) {
+        widget.grab_remove();
+        this.clearShortcut();
+        return;
+      }
+
+      if (this.keys.includes(key)) {
+        return;
+      }
+
+      this.keys.push(key);
+      this.keysEntry.text = this.keys.toString().replace(/,/g, '+');
+
+      this.buildShortcutLabelContent();
+    });
+
     // Layout
-    this.attach(modifiersLabel, 0, 0, 1, 1);
-    this.attach(this.modifiersEntry, 1, 0, 1, 1);
-    this.attach(keysLabel, 0, 1, 1, 1);
-    this.attach(this.keysEntry, 1, 1, 1, 1);
-    this.attach(repeatLabel, 0, 2, 1, 1);
-    this.attach(this.repeatSwitch, 1, 2, 1, 1);
+    this.attach(shortcutLabel, 0, 0, 1, 1);
+    this.attach(this.shortcutButton, 1, 0, 1, 1);
+    this.attach(repeatLabel, 0, 1, 1, 1);
+    this.attach(this.repeatSwitch, 1, 1, 1, 1);
     this.repeatChanged(isRepeatActive);
-    this.attach(animationLabel, 0, 4, 1, 1);
-    this.attach(this.animationCombo, 1, 4, 1, 1);
+    this.attach(animationLabel, 0, 3, 1, 1);
+    this.attach(this.animationCombo, 1, 3, 1, 1);
     this.show_all();
   }
 
   repeatChanged(isRepeatActive) {
     this.remove_row(3);
```

Comment:
This one is missing:

```suggestion
    this.remove_row(2);
```